### ISSUE TITLE
[Accordion] Foundation overrides id and aria-labelledby information

### ISSUE THREAD

**Issue by Saifadin** [zurb/foundation-sites#7923]
**How can we reproduce this bug?**
1. Create an accordion list with AngularJS ng-repeat
2. Check the id and aria-labelledby which are overridden by "panelname-label"
3. Even hardcoding it, doesnt resolve the problem

**Code Sample**

```
<div class="row">
  <div class="small-12 columns">
    <br>
    <ul class="accordion" data-accordion role="tablist" data-allow-all-closed="true">
      <li class="accordion-item" ng-repeat=" questions in FAQ ">
        <a href="#{{ questions.answerID }}" role="tab" class="accordion-title" id="{{ questions.questionID }}" aria-controls="{{ questions.answerID }}">
           <h5>{{ questions.question }}</h5>
        </a>
        <div id="{{ questions.answerID }}" class="accordion-content" role="tabpanel" data-tab-content aria-labelledby="{{ questions.questionID }}">
                {{ questions.answer }}
        </div>
      </li>
    </ul>
  </div>
</div>
```

Is there a known issue with this?

**Comment by jasonhick** [comment]
Hi,

We've just been having exactly the same problem.
We need to use the ID attribute for link tracking in Site Catalyst but it's being over written by the Foundation tabs component.

Please can this component use a different attribute.  Users shold be able to specify their own id's or at least specify what attribute to use in the config.

Kind regards,
Jason

**Comment by zurbchris** [comment]
Oversight on my part, most plugins check for an id, and if none is found, apply one where necessary. The id attribute is required for most aria attributes, which is why the js is applying on in the first place. I'll get this fixed to check first before overwriting any preexisting ids. Looks like tabs and accordion both need the check.

**Comment by zurbchris** [comment]
@Saifadin your stuff should be good. The ids being assigned are checked first here: 
```js
this.$tabs.each(function(idx, el){
      var $el = $(el),
          $content = $el.find('[data-tab-content]'),
          id = $content[0].id || Foundation.GetYoDigits(6, 'accordion'), // <-- if you already have an id applied, it won't overwrite it.
          linkId = el.id || id + '-label';
```
So the issue may be initialization order, i.e. waiting for Angular to finish loading that scope and within a digest cycle, calling `$(document).foundation()`

**Comment by zurbchris** [comment]
for tabs: c2cb63f

**Comment by Saifadin** [comment]
Thanks @zurbchris for your answer.
I am actually loading foundation in the directive after the code like this:

```
<script>$('#faqs').foundation();</script> 
```

but it seems like foundation is being loaded before that, I dont know at which point exactly

**Comment by zurbchris** [comment]
You'll probably need to use a `DOMContentLoaded` event or tap into the `$watch/$apply/$digest` callback system to wait for NG to finish it's business before you pop `$('#foo').foundation()`.

**Comment by Saifadin** [comment]
I see I'll have to try that in the morning at work, but putting that aside, if foundation generates its id-labels, shouldn't it be still working? In the rendered html tha <a> tag has id="panel1-label" and the <div> aria-labelledby="panel1-label".
It still doesnt collapse up and down

**Comment by zurbchris** [comment]
Can you create a codepen or fiddle to replicate, or at least a stack trace if there's an error? I can't think of why it wouldn't run, unless NG is swallowing the event listeners.

**Comment by Saifadin** [comment]
Yeah here you are, maybe you can help figuring it out.
https://codepen.io/anon/pen/xZPMxE

**Comment by zurbchris** [comment]
Fixed: https://codepen.io/zurbchris/pen/GoOzJd
this is the code added to your controller: 
```js
window.onload = function(){
    $(document).foundation();
  }
```
Foundation was never initialized in the example you linked.

**Comment by Saifadin** [comment]
Okay I'll try that out and come back to you guys!

**Comment by Saifadin** [comment]
Now the id i set is taken, but the accordion is still not working. It's written in the exact same way.

**Comment by Saifadin** [comment]
Here I screenshoted the generated output in Google Chrome

http://imgur.com/G3MBEJd

**Comment by Saifadin** [comment]
So after some testing with my colleagues we came to this conclusion:

It seems like the accordion is rendered correctly, but somehow the trigger for the opening and closing is not reacting (no error message in the console either).
Another thing we noticed that this code sample from the Docs is also not working although if we copy the HTML of one of the examples it is working. Something is wrong with the declarations.

```
<ul class="accordion" data-accordion role="tablist">
  <li class="accordion-item is-active">
    <!-- The tab title needs role="tab", an href, a unique ID, and aria-controls. -->
    <a href="#panel1d" role="tab" class="accordion-title" id="panel1d-heading" aria-controls="panel1d">Accordion 1</a>
    <!-- The content pane needs an ID that matches the above href, role="tabpanel", data-tab-content, and aria-labelledby. -->
    <div id="panel1d" class="accordion-content" role="tabpanel" data-tab-content aria-labelledby="panel1d-heading">
      Panel 1. Lorem ipsum dolor
    </div>
  </li>
</ul>
```

**Comment by zurbchris** [comment]
multiple calls to `$(document).foundation();` shouldn't cause problems, as it just ignores, (and logs a warning to the console), elements with previously initiated plugins.

I'm flummoxed by the thing not working though. Do me a favor in your dev build and drop this in somewhere after ng is done rendering the accordion, if you haven't done something similar already: 
```js
$('a', '.accordion').on('click', function(e){
  console.log(e.target, e.currentTarget);
});
```

I just want to verify the click handlers are firing at all, and since my codepen above works, I'm trying to debug an app remotely. I've updated my codepen to reflect the request, and it works there as well.

**Comment by Saifadin** [comment]
The click handlers are not fired at all. 

I will have to look at it later, right now we had to take the hardcoding measure in the index.html until I find a permanent solution. It is kinda weird that although i start foundation multiple times it is not triggered at all.

**Comment by Saifadin** [comment]
I think I can narrow it down now after some testing. 

The issues lays in the ng-repeat function, in a directive of AngularJS. Somehow some functions load faster then others and block each other.
It happens with the Accordion and with the Dropdown Menu. Weirdly tabs don't have this problem. I even load ```$(document).foundation();``` so often that i get a warning, that it is already loaded for some elements. 

So it's probably like this:

- directive is loaded
- foundation is initialized
- menu is generated through ng-repeat
- thats it

Hope this can help clarify

**Comment by joe-watkins** [comment]
@Saifadin Dunno if you nailed this down but perhaps you could trigger a function (which initialized Foundation) on the `$last` of `ng-repeat` ??

``` html
<li class="accordion-item" ng-repeat="questions in FAQ " ng-init="$last ? doSomething() : null">
```

**Comment by Saifadin** [comment]
I solved this problem a week ago, when it came up, by writing that part as a directive and using the controller to call $document.foundation();

The controller of a directive is called when the directive is fully loaded, thats why it works. Took a while to solve ^^

Osamah Aldoaiss
Chief Editor Midaad, Frontend Developer, Designer
Bergfeldstr. 2
38122 Braunschweig
+49 17656501994
www.aldoaiss.de

**Comment by jmathur5** [comment]
@Saifadin can you please create a codepen of your working code. I am having the exact same issue and and can't seem to get it working. Thank you so much!

**Comment by Saifadin** [comment]
> @Saifadin can you please create a codepen of your working code. I am having the exact same issue and and can't seem to get it working. Thank you so much!

@jmathur5 Sure, the final solution was, that I waited for document.ready and called it out of the controller.

See: http://codepen.io/Saifadin/pen/xqEZXO

**Comment by jmathur5** [comment]
Thank you!!!!! @Saifadin

**Comment by kball** [comment]
Trying to figure out the state of this issue... it looks like the issue of overriding values was resolved, and the remaining issue was enabling foundation after an angular render, is that correct?  Can we close this?

**Comment by JeremyEnglert** [comment]
No response. Closing.

**Comment by IamManchanda** [comment]
@Saifadin We are awaiting your reply here
If you still think there is a bug here, let us know
we will reopen

**Comment by Saifadin** [comment]
Sorry for not replying. Yes its resolved. The final solution was to wait for angular to render then enabling foundation. Thanks to @zurbchris for his help.
